hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueContainer.java

```diff
@@ -356,22 +379,25 @@ public void close() throws StorageContainerException {
       // Second sync should be a very light operation as sync has already
       // been done outside the lock.
       flushAndSyncDB();
-      updateContainerData(containerData::closeContainer);
+      updateContainerData(closer);
       clearPendingPutBlockCache();
     } finally {
       writeUnlock();
     }
-    LOG.info("Container {} is closed with bcsId {}.",
-        containerData.getContainerID(),
-        containerData.getBlockCommitSequenceId());
   }
 
-  @Override
-  public void updateDataScanTimestamp(Instant time)
+  /**
+   * For db-per-volume schemas, we don't flush and sync the whole db
+   * on closing of a single container.
+   * @param closer
+   * @throws StorageContainerException
+   */
+  private void closeWithoutFlush(Runnable closer)
```

Comment:
Can we merge closeWithFlush and closeWithoutFlush into one function, using the parameter to distinguish flush or not flush?

Reply:
Sure, that's a good idea.